File: src/main/java/recipeapplication/repository/InventoryRepository.java
package recipeapplication.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import recipeapplication.model.Ingredient;
import recipeapplication.model.InventoryItem;
import recipeapplication.model.User;

import java.util.List;
import java.util.Optional;

public interface InventoryRepository extends JpaRepository<InventoryItem, Long>
{
    List<InventoryItem> findByUser(User user);
    Optional<InventoryItem> findByIdAndUser(Long id, User user);
    Optional<InventoryItem> findByIngredientAndUser(Ingredient ingredient, User user);
    List<InventoryItem> findByIngredientDescription(String ingredient);
}

File: src/main/kotlin/recipeapplication/repository/InventoryRepository.kt
package recipeapplication.repository

import org.springframework.data.jpa.repository.JpaRepository
import recipeapplication.model.Ingredient
import recipeapplication.model.InventoryItem
import recipeapplication.model.User
import java.util.*
import javax.transaction.Transactional

interface InventoryRepository : JpaRepository<InventoryItem?, Long?> {
    fun findByUser(user: User): List<InventoryItem>
    fun findByIdAndUser(id: Long, user: User): Optional<InventoryItem>
    fun findByIngredientAndUser(ingredient: Ingredient, user: User): Optional<InventoryItem>
    fun findByIngredientDescription(ingredient: String): List<InventoryItem>

    @Transactional
    fun deleteAllByUser(user: User)
}